File: tasks/task12_1/point.py
import math

class Point:
    def __init__(self, x: float, y: float):
        self.x = x
        self.y = y

    # Distance from origin
    def distance(self) -> float:
        return math.hypot(self.x, self.y)

    # Distance to another point
    def distance_to(self, other: "Point") -> float:
        return math.hypot(self.x - other.x, self.y - other.y)


def main():
    try:
        x = float(input("Enter x coordinate: "))
        y = float(input("Enter y coordinate: "))
    except ValueError:
        print("Invalid input. Please enter valid numbers.")
        return

    p = Point(x, y)

    print(f"Distance from origin: {p.distance():.5f}")

    reference = Point(4.5, 7.0)
    print(f"Distance from (4.5, 7.0): {p.distance_to(reference):.5f}")


if __name__ == "__main__":
    main()

File: tasks/task12_1/Point.kt
import kotlin.math.hypot
class Point(var x: Double, var y: Double){
        fun distance() = hypot(x,y)
        fun distanceTo(other: Point): Double {
            return hypot(x - other.x, y-other.y)
        }
    }

fun main() {
    print("Enter x coordinate: ")
    val xInput = readLine()?.toDoubleOrNull()

    print("Enter y coordinate: ")
    val yInput = readLine()?.toDoubleOrNull()

    if (xInput == null || yInput == null) {
        println("Invalid input. Please enter valid numbers.")
        return
    }

    val p = Point(xInput, yInput)

    println("Distance from origin: %.5f".format(p.distance()))

    val reference = Point(4.5, 7.0)
    println("Distance from (4.5, 7.0): %.5f".format(p.distanceTo(reference)))
}